examples/mugen/retrieval/eval.py

```diff
@@ -4,114 +4,50 @@
-        "video_pretrain_path": {"type": str, "default": PRETRAINED_S3D_KINETICS400_URL},
-        "proj_out_dim": {"type": int, "default": 256},
-        "proj_dropout": {"type": float, "default": 0.1},
-    },
-    "evaluation": {
-        "accelerator": {"type": str, "default": "auto"},
-        "devices": {"type": int, "default": 1},
-        "checkpoint_path": {
-            "type": str,
-            "default": "lightning_videoclip_mugen_ckpt.pt",
-        },
-    },
-}
-
-
-def parse_all_args():
-    parser = argparse.ArgumentParser()
-    for arguments_dict in arg_structure.values():
-        for arg_name, arg_options in arguments_dict.items():
-            parser.add_argument("--" + arg_name, **arg_options)
-    args = parser.parse_args()
-    return args
-
-
-def get_args_from_group(args, group_name):
-    """Utility for grouping arguments based on the structure in ``arg_structure``"""
-    return argparse.Namespace(
-        **{
-            arg_name: getattr(args, arg_name)
-            for arg_name in arg_structure[group_name].keys()
-        }
-    )
+def get_yaml_config():
+    cli_conf = OmegaConf.from_cli()
+    if "config" not in cli_conf:
+        raise ValueError(
+            "Please pass 'config' to specify configuration yaml file for running VideoCLIP evaluation"
+        )
+    yaml_conf = OmegaConf.load(cli_conf.config)
+    return OmegaConf.to_container(yaml_conf)
 
 
 def evaluate():
-    args = parse_all_args()
-    datamodule_args = get_args_from_group(args, "datamodule")
-    lightningmodule_args = get_args_from_group(args, "lightningmodule")
-    dataset_args = get_args_from_group(args, "dataset")
-    videoclip_args = get_args_from_group(args, "videoclip")
-    evaluation_args = get_args_from_group(args, "evaluation")
-
-    dataset_args = MUGENDatasetArgs(get_audio=False, **vars(dataset_args))
+    args = get_yaml_config()
+    datamodule_args = args["datamodule_args"]
+    lightningmodule_args = args["lightningmodule_args"]
+    dataset_args = args["dataset_args"]
+    videoclip_args = args["videoclip_args"]
+    evaluation_args = args["evaluation_args"]
+
+    dataset_args = MUGENDatasetArgs(get_audio=False, **dataset_args)
     datamodule = MUGENDataModule(
         dataset_args,
-        text_transform=BertTextTransform(),
-        video_transform=VideoTransform(),
-        **vars(datamodule_args),
+        text_transform=BertTextTransform(**datamodule_args["bert_text_transform"]),
+        video_transform=VideoTransform(**datamodule_args["video_transform"]),
+        batch_size=datamodule_args["batch_size"],
+        num_workers=datamodule_args["num_workers"],
+        shuffle=datamodule_args["shuffle"],
     )
 
-    model = VideoCLIPLightningModule(
-        **vars(lightningmodule_args), **vars(videoclip_args)
-    )
-    model = model.load_from_checkpoint(evaluation_args.checkpoint_path)
+    model = VideoCLIPLightningModule(**lightningmodule_args, **videoclip_args)
```

Comment:
This can be error prone if the order of your expanded kwargs is different from that of your constructor signature.
Would probably map each argument explicitly here. See for example: https://github.com/facebookresearch/multimodal/blob/855c9ed7e719fab5a39b5e564073129e0e79af7d/examples/flava/finetune.py#L47-L56